src/main/kotlin/chapter12/sec4/listing.kt

```diff
@@ -160,11 +160,11 @@ val listing3 = {
         try {
             Success(SimpleDateFormat("yyyy-MM-dd").parse(dob))
         } catch (e: Exception) {
-            Failure("Data of birth must be in format yyyy-MM-dd")
+            Failure("Date of birth must be in format yyyy-MM-dd")
         }
 
     fun validPhone(phone: String): Validation<String, String> =
-        if (phone.matches("[0-9]{10}".toRegex())) Success(phone)
+        if (phone.replace("\\D", "").matches("[0-9]{10}".toRegex())) Success(phone)
```

Comment:
Not sure what this is for. Did I miss something here?

Reply:
Not at all. It's just an idea that crossed my mind when I proofed this part of the chapter...

We still check that there are exactly ten digits, but with tolerance for any punctuation we might encounter. For example, entries such as 0123456789, (0) 1 23 45 67 89, or (012) 345-6789 would pass equally the test.

The .replace("\\D", "") part I added suppresses all non-digits from the phone number to be validated.

Reply:
I feel it makes the code more difficult to understand. The focus for me is actually on how the method is used rather than what the implementation is. It is just a simple example, so mind if we bring it back to what it was?

Reply:
It's done, I just updated my branch, it will be ready to merge.